### ISSUE TITLE
Cursor Pagination Not Working - Collections not coming back as PaginatedCollections even for multiple paged results

### ISSUE THREAD

**Issue by mark-mach** [Shopify/shopify_python_api#355]
I was implementing the changes for #352 into our integration, but I cannot seem to get it to work.

Here is a code snippet:

```
        maxLimit = 2
        fparams = {"limit": maxLimit}
        prodResults = shopify.Product.find(**fparams)
        print("prodResults Instance Paginated Collection? %s" % str(isinstance(prodResults, shopify.collection.PaginatedCollection)))
        print("Product Results Instance = %s" % type(prodResults).__name__)
        for productsPage in shopify.PaginatedIterator(prodResults):
              <...>
```
And here is the output and error:

```
prodResults Instance Paginated Collection? False
Product Results Instance = Collection
Traceback (most recent call last):
  File "C:\LiClipse Workspace\Shopify\src\ToShopify\pyshopify.py", line 645, in <module>
    rs = getProducts(api_key, api_pass, api_shopname)
  File "C:\LiClipse Workspace\Shopify\src\ToShopify\pyshopify.py", line 172, in getProducts
    products = msa.getProductInfo()
  File "C:\LiClipse Workspace\Shopify\src\ToShopify\pyshopify.py", line 124, in getProductInfo
    for productsPage in shopify.PaginatedIterator(prodResults):
  File "C:\Python34\lib\site-packages\shopify\collection.py", line 131, in __init__
    raise TypeError("PaginatedIterator expects a PaginatedCollection instance")
TypeError: PaginatedIterator expects a PaginatedCollection instance

```
Here is the pip output for the ShopifyAPI:

```
C:\>pip show ShopifyAPI
---
Name: ShopifyAPI
Version: 6.0.0
Location: c:\python34\lib\site-packages
Requires: pyactiveresource, PyYAML, six
```

I know there is more than one page (especially with a limit of 2 per page), so I don't think the previous issue posted, #353 , covers the scope of this problem.  I just wanted to bring it up so that either:
A) Someone can point out where I went wrong in my implementation of #352 ; or
B) #354 will fix this issue for me as well

Thanks,

--Mark

**Comment by tanema** [comment]
Duplicate of https://github.com/Shopify/shopify_python_api/issues/353 and fixed in  https://github.com/Shopify/shopify_python_api/pull/354 released as https://github.com/Shopify/shopify_python_api/releases/tag/v6.0.1

**Comment by mark-mach** [comment]
While #354 **did** fix the exception, it **did not** fix my issue.

Here is my simplest test case (API creds removed):

```
from shopify import ShopifyResource, PaginatedIterator, Product

if __name__ == '__main__':

        apiKey = '***'
        apiPassword = '***'
        shopName = '***'
        URL = "https://%s:%s@%s.myshopify.com/admin" % (apiKey, apiPassword, shopName)
        
        ShopifyResource.set_site(URL)
        
        for productPage in PaginatedIterator(Product.find(limit=1)):
            print("Limit 1 - Product Page: ", str(productPage))
        
        for productPage in PaginatedIterator(Product.find(limit=50)):
            print("Limit 50 - Product Page: ", str(productPage))
```

This yields the results (against my test store with 3 products):

```
Limit 1 - Product Page:  [product(1012460388409)]
Limit 50 - Product Page:  [product(1012460388409), product(1012463697977), product(1012432633913)]
```

It was my understanding that PaginatedIterator was supposed to pull the automatically pull the next page of the collection when called, as stated on lines 133-138 of shopify/collection.py:

```
    >>> from shopify import Product, PaginatedIterator
    >>> for page in PaginatedIterator(Product.find()):
    ...     for item in page:
    ...         do_something(item)
    ...
    # every page and the page items are iterated
```

Please reopen this issue.

--Mark

**Comment by tanema** [comment]
The default iterator on the paginated collection will automatically fetch more. The iterator is a memory efficient and page restricted.

No iterator with auto fetch:
```python
for product in Product.find():
    do_something(product)
```    

Iterator, fetching each page (something like this, I havent tested it)
```python
cursor = shopify.Product.find()
while True:
    try:
        page = next(cursor)
        for prod in page
             do_something(prod)
    except StopIteration:
        break
```

**Comment by tanema** [comment]
@mark-mach Sorry, I understand this is confusing, I hopefully will update the docs soon

**Comment by mark-mach** [comment]
@tanema Thanks for taking a look.  I realized my test program was not including the api version to use in the URL, defaulting it to the '2019-04' version.  That version, of course, does not support cursor pagination, hence why I was only getting back a single page of results.

I updated the URL with the appropriate API version number and now PaginatedIterator works as described and as the test program expects.
